appIcons.js

```diff
@@ -1102,7 +1271,68 @@ var MyAppIcon = new Lang.Class({
     // nautilus desktop window.
     getInterestingWindows: function() {
         return getInterestingWindows(this.app, this._dtdSettings, this.monitorIndex);
-    }
+    },
+
+    insertEntryRemote: function(remote) {
+        if (!remote || this._remoteEntries.includes(remote))
```

Comment:
@azzar1 I used your new script (thanks!), and got this running on Gnome 3.22.

This line is throwing an error for me, since Debian is still on mozJS 24.

Perhaps we can change `includes` with:

```
if (!remote || this._remoteEntries.indexOf(remote) !== -1)
```